### PR TITLE
use web3 provider if it exists

### PR DESCRIPTION
**Pull request by bthaile**
if there exists provider then pass that provider down to sdk. if not then user existing ethereum http url to create provider.

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). 💎
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/vv2h1x7**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

File: packages/augur-ui/src/modules/app/actions/init-augur.ts [hunk 1]
```diff
@@ -217,54 +216,58 @@ export function connectAugur(
     if (loggedInAccountType === ACCOUNT_TYPES.TORUS) {
       preloadAccount(ACCOUNT_TYPES.TORUS);
     }
-
-    connect(
+    let provider = new JsonRpcProvider(env['ethereum-node'].http);
+    if (window.web3) {
+      provider = new Web3Provider(window.web3.currentProvider);
+    }
+    const sdk = await augurSdk.makeClient(
```
Comment (pgebheim) [inline packages/augur-ui/src/modules/app/actions/init-augur.ts]: You're in an `async` function, so you can just `await` here.

Comment (pgebheim) [reply packages/augur-ui/src/modules/app/actions/init-augur.ts]: Tom pointed out that they want this to happen in parallel with the other initialization as the app can init and then get an error where it puts up the error modal.

So this fine.

**Review by pgebheim — APPROVED** [review]

**Review by andrewdamelio — APPROVED** [review]
lgtm

**Pull request merged**